core/templates/dev/head/profile/PreferencesSpec.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Preferences Controller', function() {
+  describe('PreferencesCtrl', function() {
+    var scope, ctrl, $Http;
+
+    beforeEach(function() {
+      module('oppia');
+    });
+
+    beforeEach(inject(function(_$httpBackend_, $http, $rootScope, $controller) {
+      _$httpBackend_.expectGET('/signuphandler/data').respond({
+        username: 'myUsername',
+        has_agreed_to_latest_terms: true
+      });
+
+      $Http = $http;
+
+      mockAlertsService = {};
+
+      scope = $rootScope.$new();
+
+      ctrl = $controller('Preferences', {
+        $scope: scope,
+        $http: $http,
+        $rootScope: $rootScope,
+        alertsService: mockAlertsService
+      });
+    }));
+
+    it('should show that editor role notifications checkbox is true by default',
+      function() {
+        $Http.get(scope._PREFERENCES_DATA_URL).then(function(response) {
```

Comment:
Also, you may be interested to know that I logged scope._PREFERENCES_DATA_URL and it was undefined. So the get request is actually doing nothing, and the internal expectation is not triggering at all. That's why the test passes!

What you want is something more like: load the preferences controller and inspect the value of scope.canReceiveEditorRoleEmail(), then make sure it's the same as what was passed in via expectGET above. It's a trivial test in terms of functionality, but you might want to write it anyway, for two reasons: (i) to get familiar with how karma tests work, (ii) to establish a pattern for other developers. From the project point of view, this is optional, and it's up to you whether you'd like to do it or not; I'm happy either way.